arbor/fvm_lowered_cell_impl.hpp

```diff
@@ -67,6 +69,11 @@ class fvm_lowered_cell_impl: public fvm_lowered_cell {
         const recipe& rec,
         const fvm_discretization& D);
 
+    fvm_index_type fvm_intdom(
+        const recipe& rec,
+        const std::vector<cell_gid_type>& gids,
+        std::vector<fvm_index_type>& cell_to_intdom);
+
```

Comment:
This could do with a comment (and/or maybe a more descriptive name) indicating that it returns the number of integration domains — which might be better expressed as an `fvm_size_type` — and generates the `cell_to_intdom` map.

While essentially a private function, it is exposed for unit testing; comments here then allow us to cross check against a test implementation.

PS: Just noticed the comment on line 372; perhaps move that up here, slightly expanded?

Reply:
I added a comment describing the function and changed the type